crates/iceberg-catalog/src/service/storage/az.rs

```diff
@@ -536,72 +630,132 @@ fn is_valid_container_name(container: &str) -> Result<(), ValidationError> {
-                || c == '\''
-                || c == '('
-                || c == ')'
-                || c == ';'
-                || c == ':'
-                || c == '@'
-                || c == '&'
-                || c == '='
-                || c == '+'
-                || c == '$'
-                || c == ','
-                || c == '/'
-                || c == '?'
-                || c == '%'
-                || c == '#'
-                || c == '['
-                || c == ']'
-        }) {
-            return Err(ValidationError::InvalidProfile {
-                source: None,
-                reason:
-                    "Directory path contains reserved URL characters that are not properly escaped."
-                        .to_string(),
-                entity: "DirectoryPath".to_string(),
-            });
-        }
+fn validate_path_segment(path_segment: &str) -> Result<(), ValidationError> {
+    if path_segment.contains(|c: char| {
+        c == ' '
+            || c == '!'
+            || c == '*'
+            || c == '\''
+            || c == '('
+            || c == ')'
+            || c == ';'
+            || c == ':'
+            || c == '@'
+            || c == '&'
+            || c == '='
+            || c == '+'
+            || c == '$'
+            || c == ','
+            || c == '/'
+            || c == '?'
+            || c == '%'
+            || c == '#'
+            || c == '['
+            || c == ']'
+    }) {
+        return Err(ValidationError::InvalidProfile {
+            source: None,
+            reason:
+                "Directory path contains reserved URL characters that are not properly escaped."
+                    .to_string(),
+            entity: "DirectoryPath".to_string(),
+        });
+    }
 
-        // Check if the directory name ends with a dot (.), a backslash (\), or a combination of these.
-        if path_segment.ends_with('.') || path_segment.ends_with('\\') {
-            return Err(ValidationError::InvalidProfile {
-                source: None,
-                reason: format!(
-                    "Directory: '{path_segment}' must not end with a dot (.), or a backslash (\\)."
-                ),
-                entity: "DirectoryPath".to_string(),
-            });
+    // Check if the directory name ends with a dot (.), a backslash (\), or a combination of these.
+    if path_segment.ends_with('.') || path_segment.ends_with('\\') {
+        return Err(ValidationError::InvalidProfile {
+            source: None,
+            reason: format!(
+                "Directory: '{path_segment}' must not end with a dot (.), or a backslash (\\)."
+            ),
+            entity: "DirectoryPath".to_string(),
+        });
+    }
+
+    Ok(())
+}
+
+fn normalize_endpoint_suffix(
+    endpoint_suffix: Option<String>,
+) -> Result<Option<String>, ValidationError> {
+    // If endpoint suffix is Some(""), set it to None.
+    if let Some(endpoint_suffix) = endpoint_suffix {
```

Comment:
we could reduce nesting a bit by lifting the if let Some into the calling location and making this fn take a String instead of Option<String>

Reply:
Done - call is quite a bit uglier now but I think its worth it.